pre_commit_hooks/pretty_format_json.py

```diff
@@ -115,6 +115,17 @@ def main(argv=None):  # type: (Optional[Sequence[str]]) -> int
             if contents != pretty_contents:
                 print('File {} is not pretty-formatted'.format(json_file))
 
+                contents_by_line = ''.join(contents).split('\n')
+                pretty_contents_by_line = ''.join(pretty_contents).split('\n')
+
+                diff = len(contents_by_line) - len(pretty_contents_by_line)
+                if diff > 0:
+                    pretty_contents_by_line.extend([''] * diff)
+
+                for line_num, line in enumerate(contents_by_line):
+                    if line != pretty_contents_by_line[line_num]:
+                        print('{}:{}'.format(json_file, line_num))
```

Comment:
probably just use `difflib` to print a diff instead

could you add a test as well?

Reply:
Any advice on how to test? From what I can see, these changes don't change the return value of the function.

Do you think it would be more beneficial to put a diff behind a flag? That way it's more easily tested, and it doesn't alter default behavior?

Reply:
You can use the [`capsys`](https://docs.pytest.org/en/latest/capture.html#accessing-captured-output-from-a-test-function) pytest fixture -- there's a few examples in some of the tests already iirc

I would only print the diff in the case of the non-auto-fixing mode (so it's already behind a flag)

Reply:
Ok, will look in a bit - thanks!